src/utils/fetchCustomPaperDefinitions.js

```diff
@@ -0,0 +1,44 @@
+import { Q, fetchPolicies } from 'cozy-client'
+
+import { FILES_DOCTYPE } from 'src/doctypes'
+import getOrCreateAppFolderWithReference from 'src/helpers/getFolderWithReference'
+
+const paperConfigFilenameCustom = 'papersDefinitions.json'
+
+/**
+ * Fetch custom PaperDefinitions file in "My papers" folder (Drive)
+ * @param {CozyClient} client - Instance of CozyClient
+ * @param {Function} t - i18n function
+ * @returns {Promise<{ paperConfigFilenameCustom: string, appFolderPath: string, data: object|null }>}
+ */
+export const fetchCustomPaperDefinitions = async (client, t) => {
+  let appFolderPath = ''
+  try {
+    const { _id: appFolderId, path } = await getOrCreateAppFolderWithReference(
+      client,
+      t
+    )
+    appFolderPath = path
+    const fileColl = client.collection(FILES_DOCTYPE)
+    const queryDef = Q(FILES_DOCTYPE)
+      .where({ dir_id: appFolderId, name: paperConfigFilenameCustom })
+      .partialIndex({ trashed: false })
+      .indexFields(['name', 'dir_id'])
+
+    const { data } = await client.query(queryDef, {
+      as: `fetchJsonFileByName`,
+      fetchPolicy: fetchPolicies.noFetch()
+    })
+
+    if (data?.length > 0) {
+      const dataBin = await fileColl.fetchFileContentById(data[0]._id)
+      const dataJSON = await dataBin.json()
+
+      return { paperConfigFilenameCustom, appFolderPath, data: dataJSON }
+    }
+
+    return { paperConfigFilenameCustom, appFolderPath, data: null }
+  } catch (error) {
+    return { paperConfigFilenameCustom, appFolderPath, data: null }
+  }
+}
```

Comment:
fetch should only fetch, the translation should not be handled here

Reply:
Indeed, it's corrected 👍